dhall-json/src/Dhall/JSONToDhall.hs

```diff
@@ -344,44 +359,45 @@ module Dhall.JSONToDhall (
     , showCompileError
     ) where
 
-import           Control.Applicative ((<|>))
-import           Control.Exception (Exception, throwIO)
-import           Control.Monad.Catch (throwM, MonadCatch)
-import           Data.Aeson (Value)
-import qualified Data.Aeson as A
-import           Data.Aeson.Encode.Pretty (encodePretty)
+import Control.Applicative      ((<|>))
+import Control.Exception        (Exception, throwIO)
+import Control.Monad.Catch      (MonadCatch, throwM)
+import Data.Aeson               (Value)
+import Data.Aeson.Encode.Pretty (encodePretty)
+import Data.Either              (rights)
+import Data.Foldable            (toList)
+import Data.List                ((\\))
+import Data.Monoid              (Any (..))
+import Data.Scientific          (floatingOrInteger, toRealFloat)
+import Data.Semigroup           (Semigroup (..))
+import Data.Text                (Text)
+import Data.Void                (Void)
+import Dhall.Core               (Chunks (..), DhallDouble (..), Expr (App))
+import Dhall.JSON.Util          (pattern V)
+import Dhall.Parser             (Src)
+import Options.Applicative      (Parser)
+
+import qualified Data.Aeson                 as A
+import qualified Data.Aeson.Types           as AT
```

Comment:
(Style, feel free to ignore)

I often can't guess what an abbreviation like `A` or `AT` stands for, so I prefer fully qualified modules or at least slightly longer names:

```suggestion
import qualified Data.Aeson                 as Aeson
import qualified Data.Aeson.Types           as Aeson.Types
```